programs/cleos/main.cpp

```diff
@@ -4279,7 +4279,7 @@ int main( int argc, char** argv ) {
          try {
             priv_key = private_key_type(str_private_key);
          } EOS_RETHROW_EXCEPTIONS(private_key_type_exception, "Invalid private key")
-         trx.sign(priv_key, *chain_id);
+         trx.sign(priv_key, *chain_id, fc::require_canonical_t::yes);
```

Comment:
Should we consider making `canonical` an option to `cleos sign` ?
What about `keosd`, should `keosd` have a `config.ini` option to sign `canonical` or not?

Reply:
What about we discuss this in standup and create another issue if we decide to do it. I feel it doesn't have to be included in this PR, which is already plenty big.

Reply:
https://github.com/AntelopeIO/spring/issues/1757